### PR TITLE
Add a switch to use redfish as BMC driver

### PR DESCRIPTION
**Pull request by kashifest**
Add a switch for using redfish bmc driver

### CODE REVIEW

File: config_example.sh [hunk 1]
```diff
@@ -74,3 +74,8 @@
 # Number of maximum verification or test retries
 #
 #export TEST_MAX_TIME=120
+
+#
+# Set the driver
+#
+#export BMC_DRIVER="redfish"
```
Comment (maelk) [inline config_example.sh]: we should maybe add a note that we use ipmi by default. or switch to redfish by default ?

File: vm-setup/roles/libvirt/tasks/vm_setup_tasks.yml [hunk 2]
```diff
@@ -112,6 +112,17 @@
       when: item is failed
       with_items: "{{ vm_extradisks_check.results }}"
 
+    - name: Get vm uuid
```
Comment (maelk) [inline vm-setup/roles/libvirt/tasks/vm_setup_tasks.yml]: maybe a comment to say why we need the uuid for redfish ? (mapped to system-id)

File: config_example.sh (hunk 1 above)
Comment (kashifest) [reply config_example.sh]: Done

File: vm-setup/roles/libvirt/tasks/vm_setup_tasks.yml (hunk 2 above)
Comment (kashifest) [reply vm-setup/roles/libvirt/tasks/vm_setup_tasks.yml]: Done

File: vm-setup/roles/libvirt/templates/ironic_nodes.json.j2 [hunk 3]
```diff
@@ -12,13 +12,18 @@
...
-        "ipmi_address": "{{ lvars['host_ip'] }}",
-        "ipmi_port": "{{ node.virtualbmc_port }}",
+        "username": "admin",
+        "password": "password",
+        {% if vm_driver =='redfish' -%}
+        "port": "8000",
+        "address": "{{vm_driver}}+http://{{ lvars['host_ip'] }}:8000/redfish/v1/Systems/{{vm_id[node.name]}}",
+        {% else -%}
+        "port": "{{ node.virtualbmc_port }}",
+        "address": "{{vm_driver}}://{{lvars['host_ip']}}:{{node.virtualbmc_port}}",
```
Comment (hardys) [inline vm-setup/roles/libvirt/templates/ironic_nodes.json.j2]: I wonder if we still need the "port" key, if it's going to be included in the uri value of "address" now?

Comment (kashifest) [reply vm-setup/roles/libvirt/templates/ironic_nodes.json.j2]: Agreed. Just kept it there for better visibility

**Comment by kashifest** [comment]
@russellb what are your thoughts on this PR?  I cannot add you as a reviewer

**Comment by russellb** [comment]
To add a reviewer, you can comment `/cc russellb`

**Comment by russellb** [comment]
@hardys you OK if we merge this along with your fixes to the openshift dev-scripts repo?

**Comment by hardys** [comment]
> @hardys you OK if we merge this along with your fixes to the openshift dev-scripts repo?

Yes that's fine by me, and I've mentioned the json format change to those who may be impacted downstream of this change (e.g those using NODES_FILE in openshift-metal3/dev-scripts to on deploy baremetal).

**Pull request merged**